cranelift/codegen/src/isa/unwind/winx64.rs

```diff
@@ -285,6 +324,71 @@ impl UnwindInfo {
     }
 }
 
+const UNWIND_RBP_REG: u8 = 5;
+
+pub(crate) fn create_unwind_info_from_insts<MR: RegisterMapper<regalloc::Reg>>(
+    insts: &[(CodeOffset, UnwindInst)],
+) -> CodegenResult<UnwindInfo> {
+    let mut unwind_codes = vec![];
+    let mut frame_register_offset = 0;
+    let mut max_unwind_offset = 0;
+    for &(offset, ref inst) in insts {
+        match inst {
+            &UnwindInst::PushOldFP { .. } => {
+                let offset = ensure_unwind_offset(offset)?;
+                unwind_codes.push(UnwindCode::PushRegister {
+                    offset,
+                    reg: UNWIND_RBP_REG,
+                });
+            }
+            &UnwindInst::PopOldFP { .. } | &UnwindInst::Ret => {
+                // Don't count as part of the prologue. Otherwise the SEH info will be too large
+                // (it can only cover 256 bytes of code).
+                continue;
+            }
+            &UnwindInst::CreateFPFrame { fp_offset } => {
+                let offset = ensure_unwind_offset(offset)?;
+                frame_register_offset = fp_offset;
+                unwind_codes.push(UnwindCode::SetFPReg { offset });
+                if fp_offset > 0 {
+                    unwind_codes.push(UnwindCode::StackAlloc {
```

Comment:
Perhaps it should just be called `CreateFrame` -- it was meant to be a single macro-op that captured anything a specific unwind format needs for the prologue. The close mapping between `sub %rsp` and `StackAlloc` was the sort of thing I was hoping to avoid in this simplification :-) Given that we pin the frame to `RBP` with the `SetFPReg`, the `StackAlloc` serves only to let the unwinder know that more stack space is being used; so as long as it comes after the `SetFPReg` and before any uses of the clobber space, it shouldn't matter exactly where it occurs, I think.

Reply:
Ok I think that's fine in the meantime anyway. I don't think it's required we get this 100% right but it might just be worth in the documentation to mention that it's sort of a macro-op for now.

Reply:
I did one better and just removed the `StackAlloc`; it doesn't seem to actually be needed once we pin the frame to `RBP`.